qtpynodeeditor/node_connection_interaction.py

```diff
@@ -101,7 +100,7 @@ def can_connect(self) -> bool:
 
         candidate_node_data_type = port.data_type
         if connection_data_type.id == candidate_node_data_type.id:
-            return port, DefaultTypeConverter
+            return port, None
```

Comment:
I don't know if the default one should be returned if the data types are equal, that's up to you @klauer. For now I did it like this because in `FlowScene.restore_connection` the `return self._registry.get_type_converter(out_type, in_type)` returns `None` and not the default converter. However, one can turn this around and make the registry return the default converter instead of `None` if nothing has been registered. Anyway, the commit before this tests that the behavior is consistent.

Reply:
I had the same dilemma when I was changing the code. Back then I thought that if the types are the same, then we don't need any converter, so the straightforward solution was `None`. But I also think it would be a bit cleaner to use the default one. In that case one could probably save some if-else statements when doing the conversion in the pipeline.

Reply:
Yeah, I can see both sides of the argument here. Given that `None` is already implemented, we can just go forward with it now, unless you have some extra time to circle back to it.